.azure-pipelines/pipelines.yml

```diff
@@ -0,0 +1,29 @@
+trigger:
+  branches:
+    include:
+    - "main"
+
+stages:
+- stage: salvo_test_and_check
+  dependsOn: []
+  pool: "x64-large"
+  jobs:
+  - job: salvo_test_and_check
+    displayName: "do_ci.sh"
+    dependsOn: []
+    strategy:
+      maxParallel: 4
```

Comment:
Sorry, when configured like this, the parallel jobs actually request 4 VMs and execute in parallel on them. That seems like an unnecessary amount of resources for Salvo build / check / test steps.

https://dev.azure.com/cncf/envoy/_build/results?buildId=106913&view=results

We will want to have dedicated VMs for performance tests, but we can save some resources here. Can we refactor this so that these jobs either run serially on 1 VM at a time, or one after each other on the same VM?

Reply:
@mum4k Correct me I was wrong, I failed to find a way in AZP to run above parallel 4 steps in one VM unless we add one new `CI_TARGET` like 'all' which will execute all steps. What do you think?

Reply:
I think there are two ways to achieve this:

1. Define these as 4 jobs that depend on each other, i.e. the next job only starts after the first one finishes. See the `dependsOn` list.
1. What you mentioned - add one CI target that executes all the steps in one script.

Approach (1) will take longer, because it will set up a VM for each of the jobs, but it will also execute hermetically and will only use one VM at a time. If it won't take too long to run it this way, I think we should prefer (1). Can you give it a try?

Reply:
(fyi) Configuring (1) might be as simple as setting `maxParallel: 1`.

Reply:
> Define these as 4 jobs that depend on each other, i.e. the next job only starts after the first one finishes. See the dependsOn list.

Correct me if it's wrong. I tried defining them depend on each other like this: 
```
    strategy:
      maxParallel: 1
      matrix:
        build:
          CI_TARGET: "build"
        format:
          dependsOn: build
          CI_TARGET: "check_format"
        test:
          dependsOn: test
          CI_TARGET: "test"
        coverage:
          dependsOn: coverage
          CI_TARGET: "coverage"
```
however, it seems that it still created 4 VMs, you can find those agent machines' names here:
https://dev.azure.com/cncf/envoy/_build/results?buildId=106923&view=logs&j=1624a51a-a39c-5947-4081-f6b573564ebe&t=f59bbc51-ac17-4b90-a598-d95b3f333e77&l=3
https://dev.azure.com/cncf/envoy/_build/results?buildId=106923&view=logs&j=27dc7755-952e-5a2b-a136-5632fca0d844&t=2dae8299-bfbc-417e-8ecc-06c2aec77099&l=3
https://dev.azure.com/cncf/envoy/_build/results?buildId=106923&view=logs&j=45e0cd60-018b-5371-3188-0d384cbaa76f&t=ac498b66-e74e-472f-a607-2988e9c239ea&l=3
https://dev.azure.com/cncf/envoy/_build/results?buildId=106923&view=logs&j=35882957-0705-54d1-237e-39e046a21a26&t=6741b2c4-c2af-40c2-9239-5bdec56e2c57&l=3

Reply:
Sorry for misunderstanding what you said, yes, though approach (1) will create 4 VMs, it will use one VM at the same time. And we the simplest way to implement it is setting `maxParallel: 1`